arcgis-ios-sdk-samples/Analysis/Viewshed (GeoElement)/README.md

```diff
@@ -1,35 +1,43 @@
 
-Once the scene is done loading, click on a location for the tank to drive to. It will automatically turn and drive straight towards the clicked point. The viewshed will automatically move and rotate with the tank.
+## How to use the sample
 
-![](image1.png)
+Tap to set a destination for the vehicle (an  `AGSGeoElement`). The vehicle will 'drive' towards the tapped location. The viewshed analysis will update as the vehicle moves.
 
 ## How it works
 
-<p>To attach a viewshed to an <code>AGSGeoElement</code>:</p>
-
-<ol>
-  <li>Create an <code>AGSGraphic</code> and add it to an <code>AGSGraphicsOverlay</code>.</li>
-  <li>Use an <code>AGSSimpleRenderer</code> in the <code>AGSGraphicsOverlay</code> which has a heading expression set. This 
-  way you can relate the viewshed's heading to the <code>AGSGeoElement</code>'s heading.
-  <li>Create an <code>AGSGeoElementViewshed</code> with the graphic, heading/pitch offsets, and min/max distance.</li>
-  <li>To offset the viewshed's observer location from the center of the graphic, set <code>viewshed.offsetX</code>, etc.</li>
-</ol>
-
-<h2>Features</h2>
-
-<ul>
-  <li>3D</li>
-  <li>AGSAnalysisOverlay</li>
-  <li>AGSArcGISTiledElevationSource</li>
-  <li>AGSArcGISScene</li>
-  <li>AGSArcGISSceneLayer</li>
-  <li>AGSGeoElementViewshed</li>
-  <li>AGSGraphic</li>
-  <li>AGSSceneView</li>
-</ul>
+1. Create and show the scene, with an elevation source and a buildings layer.
+2. Add an `AGSModelSceneSymbol` (the GeoElement) to represent the observer (in this case, a tank).
+    * Use an `AGSSimpleRenderer` which has a heading expression set in the `AGSRendererSceneProperties`. This way you can relate the viewshed's heading to the object's heading.
+3. Create an `AGSGeoElementViewshed` with configuration for the viewshed analysis.
+4. Add the viewshed to an `AGSAnalysisOverlay` and add the overlay to the scene.
+5. Configure the SceneView `AGSCameraController` to orbit the vehicle.
+
+## Offline data
+
+[Model Marker Symbol Data](https://www.arcgis.com/home/item.html?id=07d62a792ab6496d9b772a24efea45d0) will be downloaded by the sample viewer automatically.
+
+## About the data
+
+This sample shows a [Johannesburg, South Africa Scene](https://www.arcgis.com/home/item.html?id=eb4dab9e61b24fe2919a0e6f7905321e) from ArcGIS Online. The sample uses a [Tank model scene symbol](https://www.arcgis.com/home/item.html?id=07d62a792ab6496d9b772a24efea45d0) hosted as an item on ArcGIS Online.
+
+## Relevant API
+
+* AGSAnalysisOverlay
+* AGSGeodeticDistanceResult 
+* AGSGeoElementViewshed
+* class AGSGeometryEngine.geodeticDistanceBetweenPoint1(_:point2:distanceUnit:azimuthUnit:curveType:) (used to animate the vehicle)
```

Comment:
I put all class methods at top, so that it looks better... 😅 Doesn't matter though.

Reply:
Decided to stick to just the class to simplify and standardize things!